app/preview.py

```diff
@@ -65,6 +65,27 @@ def page_count():
     )
 
 
+@statsd(namespace="template_preview")
+def get_pdf_redis_key(json):
+    def print_dict(d):
+        """
+        From any environment/system will return the same string representation of a given dict.
+
+        ASSUMPTION: all dict values are immutable (eg int, none, str)
+        """
+        return sorted(d.items())
+
+    unique_name_dict = {
+        'template_id': json['template']['id'],
+        'version': json['template']['version'],
+        'dvla_org_id': json['dvla_org_id'],
+        'letter_contact_block': json['letter_contact_block'],
+        'values': None if not json['values'] else print_dict(json['values'])
+    }
+
+    return print_dict(unique_name_dict)
```

Comment:
this currently returns a list - should we cast this to a string at this point?

Reply:
Not changed. Is there a benefit to doing that?

Reply:
hmm, if redis is using it as a string then just clarity i guess. it's probably fine here i spose

Reply:
Would it work if we use sorted(d.values()) rather than items?

Reply:
if someone uses the same placeholders in a different order while previewing it'd not be able to distinguish between them - so you could get the wrong PDF returned